code.ts

```diff
@@ -1,10 +1,49 @@
 const delay = ms => new Promise(res => setTimeout(res, ms));
 var activeTimer = 0;
 const secondsSet = [86400, 3600, 60, 1];
+const progressBar = figma.createRectangle();
+var pause = 0;
+var reset = 0;
+var userSetSeconds = 0;
 
-if (checkForSelectedNodes() == false) {
-  if (checkForNodesThatBeginWithTimer() == false) {
-    throw new Error("Type the time to start Timer");
+figma.showUI(__html__, { width: 220, height: 50 })
+
+figma.ui.onmessage = msg => {
+ 
+  if (msg.type === 'start') {
+    pause = 0;
+    reset = 0;
+    checkAndStart();
+  }
+
+  if (msg.type === 'pause') {
+    pause = 1;
+  }
+
+  if (msg.type === 'continue') {
+    pause = 0;
+  }
+
+  if (msg.type === 'reset'){
+    reset = 1;
+    pause = 1;
+  }
+
+  if (msg.type === 'helpon'){
+    figma.ui.resize(220,150);
+  }
+
+  if (msg.type === 'helpoff'){
+    figma.ui.resize(220,50);
+  }
+};
+
+function checkAndStart()
+{
```

Comment:
small formatting improvement:
```suggestion
function checkAndStart() {
```